web/src/user_group_edit.ts

```diff
@@ -154,8 +192,10 @@ function show_membership_settings(group) {
     update_members_panel_ui(group);
 }
 
-function show_general_settings(group) {
-    const permission_settings = Object.keys(realm.server_supported_permission_settings.group);
+function show_general_settings(group: UserGroup): void {
+    const permission_settings = z
+        .array(group_setting_name_schema)
+        .parse(Object.keys(realm.server_supported_permission_settings.group));
```

Comment:
This parsing logic should probably be moved into `web/src/group_permission_settings.ts`, which could export the type-checked object. Can be a follow-up, maybe by @sahil839 as settings maintainer.